README.md

```diff
@@ -288,6 +288,78 @@ const req = {
 const sig = await client.sign(req)
 ```
 
+### 💾 Calldata Decoding
+
+> NOTE: all available calldata decoding options will be documented in this section. More may be added as time goes on.
+
+Certain transaction decoder types may support calldata decoding for request data. You can use this feature by including "calldata decoder data" (explained shortly) in a general signing request using the `decoder` request param:
+
+```
+req.decoder = <calldata decoder data>
+await client.sign(req);
+```
+
+If you include a valid calldata decoder, the appearance of the transaction's data on the user's Lattice should transform from a raw hex string to a markdown-style version which displays the function name, parameter names, and values.
+
+**Storing Calldata Decoders**
+
+Although not necessary, in certain situations it may be advantageous to pre-save decoders to the Lattice. One advantage is that if the decoder is saved, you do not need to include it in the transaction request, which frees up some space. Additionally, pre-saving data may unlock certain security features depending on the decoder type.
+
+You can use the following API:
+
+> Please see API docs for all options. Also see tests in `test/signing/evm.ts` for examples on usage.
+
+* `addDecoders`: Allows the user to add a series of calldata decoders for a specific decoder type (e.g. EVM). This will draw a screen on the user's Lattice which will require approval before returning success.
+* `getDecoders`: Fetch `n` consecutive decoders for a specific type, starting a specific index. Does not draw a screen.
+* `removeDecoders`: Remove a set of included decoders for a specific type. Does not draw a screen. You can also set a flag to remove all decoders for a specific type.
+
+#### 1️⃣  EVM
+
+EVM transactions serialize calldata according to the [Ethereum ABI specification](https://docs.soliditylang.org/en/latest/abi-spec.html). The first four bytes of a transaction's `data` represent the "function selector", which is (sort of) a unique identifier for a given function. You can build the calldata decoder data by either parsing a [Solidity JSON ABI](https://docs.ethers.io/v5/api/utils/abi/formats/#abi-formats--solidity) object (which you can fetch from [Etherscan](https://etherscan.io)) or by parsing an ABI canonical name (you can get this from [4byte](https://www.4byte.directory)). *Using the Solidity JSON ABI is recommended*.
+
+> Note: We do not support 100% of all edge cases in the ABI specification, but we do support the vast majority of types
```

Comment:
```suggestion
> Note: We do not support 100% of all edge cases in the ABI specification, but we do support the vast majority of types. Please open a PR or an issue if your case is not yet supported.
```

Maybe a good spot for a call to action?

Reply:
[756bc00](https://github.com/GridPlus/gridplus-sdk/pull/353/commits/756bc001e62e4ba9fce75734645727253137ddc2)